### PR TITLE
Enhancements :: Table bugfixes and option for pagination number added

### LINKED ISSUE #1926: Filtered results does not get  downloaded.
### What is the expected behaviour?
Whenever the filter is applied on the table widget and saved then download, the downloaded file should display the filtered results.

### What is the current behaviour?
Whenever the filter is applied on the table widget and saved then download, it downloads the original data in the table.

### How to reproduce the issue?

1. Drag and drop the table to the canvas.
2. Click on filter.
3. Add filter as id greater than 1.
4. Save the filter.
5. Download the result & observe.

### Screenshots or Screencast
https://drive.google.com/file/d/1lny4b3dIGJIBz_g6uO43o4utFYVCfM-e/view?usp=sharing

### LINKED ISSUE #4082: Enhancements table widget
### Describe the solution you'd like

*resizing columns lead to sorting : bug
*if filter is applied- download filtered data only : bug
*add option to change number of results per page - client side pagination
*set size of table to always full width

### PR DESCRIPTION
**Pull request by stepinfwd**
close #4082
closes #1926
**4 clickup tasks are linked**

### CODE REVIEW

**Comment by sherfin94** [comment]
@stepinfwd There is still a very small gulf between the right most border of the table and the right most border of the right most column: 
image

This might also be causing the horizontal scrollbar to appear all the time. That is, there seems to be no way to get rid of the horizontal scroll bar at all.

**Comment by sherfin94** [comment]
`Number of results per page` is being shown even if it is not client side pagination. Please show it only for client side pagination, and try to see if we can show it below the client side pagination option. Also, lets call it `Number of rows per page`
image

File: frontend/src/_styles/theme.scss [hunk 1]
```diff
@@ -6355,3 +6355,27 @@ input.hide-input-arrows{
     height: 100%;
   }
 }
+// Table set to full width
+.jet-data-table thead {
+  display: flex !important;
```
Comment (sherfin94) [inline frontend/src/_styles/theme.scss:6360]: This is not urgent, but see if we could avoid `!important` if possible.

**Review by sherfin94 — CHANGES_REQUESTED** [review]
I've commented few issues.

Implementation generally looks good!

**Review by sherfin94 — CHANGES_REQUESTED** [review]
Hey @stepinfwd ,

Few items:

1. Resizing still seems to trigger sort in certain scenarios: 

https://user-images.githubusercontent.com/20333972/192452090-7c6ad8c6-ebea-4172-b4ed-b596786af116.mov


https://user-images.githubusercontent.com/20333972/192452131-f7092158-dc89-4056-b180-f716d772ff38.mov

2. If it is a server side pagination, lets not show the number of records per page property: 
image

3. Resolve conflicts

**Comment by shubh22** [comment]
Task linked: [CU-2vjw3h2 Table Enhancements](https://app.clickup.com/t/2vjw3h2)

**Comment by shubh22** [comment]
Task linked: [CU-2q9mdw7 Resizing column, automatically sorts the column in table](https://app.clickup.com/t/2q9mdw7)

**Comment by shubh22** [comment]
Task linked: [CU-2vjy6xc Downloaded data must be according  filter - Table widgets](https://app.clickup.com/t/2vjy6xc)

**Comment by shubh22** [comment]
Task linked: [CU-2vjy6y0 Change the number of results per page - client side pagination (configurable by user)-Table widget](https://app.clickup.com/t/2vjy6y0)

**Review by sherfin94 — APPROVED** [review]

**Pull request merged**